events/events.go

```diff
@@ -161,40 +164,102 @@ func (em *EventManager) Subscribe(ctx context.Context, ident string, filter func
 		broadcastCounter: eventsBroadcast.WithLabelValues(ident),
 	}
 
+	cleanup := func() {
+		close(done)
+		em.rmSubscriber(sub)
+	}
+
+	if since == nil {
+		em.addSubscriber(sub)
+		return sub.outgoing, cleanup, nil
+	}
+
+	out := make(chan *XRPCStreamEvent, em.bufferSize)
+
 	go func() {
-		if since != nil {
-			if err := em.persister.Playback(ctx, *since, func(e *XRPCStreamEvent) error {
-				select {
-				case <-done:
-					return ErrPlaybackShutdown
-				case sub.outgoing <- e:
-					return nil
-				}
-			}); err != nil {
-				if errors.Is(err, ErrPlaybackShutdown) {
-					log.Warnf("events playback: %s", err)
-				} else {
-					log.Errorf("events playback: %s", err)
+		lastSeq := *since
+		// run playback to get through *most* of the events, getting our current cursor close to realtime
+		if err := em.persister.Playback(ctx, *since, func(e *XRPCStreamEvent) error {
```

Comment:
in my 'ringbuf' persister implement in this PR, I take care to get very close to 'live'. 
That is not necessarily (yet) the case for the disk persister, and im not sure where i want that responsibility to lie. Should the caller just call it until they stop getting new events? or should the persister try to return all events it knows about up until the point that it returns?

Reply:
I think it might make sense for the persister to make an effort to playback more events but I don't think it should aim for 100% if it's too complex. Maybe it would make sense to check the logfile refs if we're on the last logfile we know about and then keep looping if there are more files now until there aren't. If we had larger logfiles I'd be less concerned about the Persister getting stuck trying to catch a user up since 100k event logfiles would take a lot longer to roll over etc.

Reply:
I think the aim for the DiskPersister should be to playback until the user is in the _current_ logfile though, that shouldn't be _too_ hard.